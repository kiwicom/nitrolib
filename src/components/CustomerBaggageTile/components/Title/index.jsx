// @flow strict
import * as React from "react";
import styled, { css } from "styled-components";
import GenderMan from "@kiwicom/orbit-components/lib/icons/GenderMan";
import GenderWoman from "@kiwicom/orbit-components/lib/icons/GenderWoman";
import Stack from "@kiwicom/orbit-components/lib/Stack";

import type { Price, Gender, OrderStatusType } from "../../../../records/Baggage";
import mq from "../../../../styles/mq";
import Badge from "./components/Badge";

type TitleProps = {
  firstName: string,
  lastName: string,
  gender: Gender,
  orderStatus: OrderStatusType,
  price?: Price,
};

const TitleWrapper = styled.div`
  display: flex;
  align-items: center;
  ${mq.ltBigMobile(css`
    align-items: flex-start;

    flex-direction: column;
  `)};
  svg {
    margin-right: 6px;
  }

  div {
    margin-left: 10px;
    ${mq.ltBigMobile(css`
      margin-top: 9px;
      margin-left: 28px;
    `)};
  }
`;

const PassengerName = styled.span`
  margin-right: 16px;
`;

const Title = ({ firstName, lastName, gender, orderStatus, price }: TitleProps) => (
  <TitleWrapper>
    <Stack inline align="center" spacing="condensed">
      {gender === "male" ? <GenderMan /> : <GenderWoman />}
      <PassengerName>{`${firstName} ${lastName}`}</PassengerName>
    </Stack>
    <Badge orderStatus={orderStatus} price={price} />
  </TitleWrapper>
);

export default Title;
