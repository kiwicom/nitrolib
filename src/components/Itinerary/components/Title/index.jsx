// @flow strict
import * as React from "react";
import styled from "styled-components";
import FlightDirection from "@kiwicom/orbit-components/lib/icons/FlightDirect";
import Stack from "@kiwicom/orbit-components/lib/Stack";
import { left } from "@kiwicom/orbit-components/lib/utils/rtl";

import TranslateNode from "../../../TranslateNode";
import Time from "../../../Time";
import tripColors from "../../services/tripColors";
import { themeDefault } from "../../../../records/Theme";
import type { ThemeProps } from "../../../../records/Theme";
import Translate from "../../../Translate";

type Props = {|
  sourceName: string,
  destinationName: string,
  duration: number,
  index?: number,
  type: "oneWay" | "return" | "multicity" | "nomad",
  direction: "outbound" | "inbound",
|};

const DIRECTION_TYPES = {
  inbound: __("result.inbound"),
  outbound: __("result.outbound"),
};

const TripLabel = styled.div`
  color: ${({ tripIndex }) => tripColors(tripIndex + 1, "#000")};
  font-weight: ${({ theme }: ThemeProps) => theme.orbit.fontWeightBold};
  text-transform: uppercase;
  font-size: ${({ theme }: ThemeProps) => theme.orbit.spaceSmall};
`;

TripLabel.defaultProps = {
  theme: themeDefault,
};

const TripTrime = styled.div`
  font-size: ${({ theme }: ThemeProps) => theme.orbit.spaceSmall};
  padding-${/* sc-custom "left" */ left}: ${({ theme }: ThemeProps) => theme.orbit.spaceXSmall};
`;

TripTrime.defaultProps = {
  theme: themeDefault,
};

const TitleWrapper = styled.div`
  padding-top: ${({ theme }: ThemeProps) => theme.orbit.spaceLarge};
  margin-bottom: ${({ theme }: ThemeProps) => theme.orbit.spaceMedium};
  display: flex;
  align-items: center;
`;

TitleWrapper.defaultProps = {
  theme: themeDefault,
};

const ItineraryTripTitle = ({
  sourceName,
  destinationName,
  duration,
  type,
  direction,
  index,
}: Props) => (
  <TitleWrapper>
    <TripLabel tripIndex={index}>
      {type === "multicity" || type === "nomad" ? (
        <Stack align="center" spacing="tight">
          {sourceName}
          <FlightDirection />
          {destinationName}
        </Stack>
      ) : (
        direction && <Translate t={DIRECTION_TYPES[direction]} />
      )}
    </TripLabel>
    <TripTrime>
      <TranslateNode
        t="common.duration"
        values={{
          duration: <Time time={new Date(duration)} />,
        }}
      />
    </TripTrime>
  </TitleWrapper>
);

export default ItineraryTripTitle;
