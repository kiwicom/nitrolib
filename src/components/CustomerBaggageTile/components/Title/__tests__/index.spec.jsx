// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import Title from "../index";

const props = {
  firstName: "Oliver",
  middleName: "Olix",
  lastName: "Dlouhy",
  gender: "male",
  orderStatus: "unpaid",
  dayOfBirth: undefined,
  price: 21,
};

describe("#Title", () => {
  test("renders", () => {
    const wrapper = shallow(<Title {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders proper gender icon", () => {
    const wrapper = shallow(<Title {...props} gender="female" />);
    expect(wrapper.find("GenderWoman").exists()).toBe(true);
  });
});
