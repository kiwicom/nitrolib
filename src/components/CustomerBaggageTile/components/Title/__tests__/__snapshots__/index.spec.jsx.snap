// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Title renders 1`] = `
<Title__TitleWrapper>
  <Stack
    align="center"
    inline={true}
    spacing="condensed"
  >
    <GenderMan />
    <Text
      element="span"
      size="large"
      weight="bold"
    >
      Oliver
        Olix
        Dlouhy
    </Text>
  </Stack>
  <TitleBadge
    orderStatus="unpaid"
    price={21}
  />
</Title__TitleWrapper>
`;
